File: android/edge-frameworks-core/src/main/kotlin/io/github/rajibsarwar/edgeframeworks/EdgeContentFingerprint.kt
package io.github.rajibsarwar.edgeframeworks

import java.security.MessageDigest

object EdgeContentFingerprint {
    fun sha256(
        bytes: ByteArray
    ): String {
        return MessageDigest
            .getInstance("SHA-256")
            .digest(bytes)
            .joinToString("") {
                "%02x".format(
                    it.toInt() and 0xff
                )
            }
    }

    fun sha256(
        text: String
    ): String {
        return sha256(
            text.toByteArray(Charsets.UTF_8)
        )
    }
}

File: ios/Sources/EdgeFrameworks/Runtime/EdgeContentFingerprint.swift
import CryptoKit
import Foundation

public enum EdgeContentFingerprint {
    public static func sha256(
        _ data: Data
    ) -> String {
        SHA256.hash(data: data)
            .map {
                String(
                    format: "%02x",
                    $0
                )
            }
            .joined()
    }

    public static func sha256(
        _ text: String
    ) -> String {
        sha256(
            Data(text.utf8)
        )
    }
}
